### ISSUE TITLE
CA1416: is only supported on: 'windows' 5.1.2600 and later

### ISSUE THREAD

**Issue by myokeeh** [microsoft/CsWin32#1147]
Using 

```
_ = PInvoke.CreateHardLink(pcwstr_pHardlink_filename, pcwstr_pPhoto_filename, null);
```
and getting the following warning:

```
Warning	CA1416	This call site is reachable on: 'windows' all versions. 'PInvoke.CreateHardLink(PCWSTR, PCWSTR, SECURITY_ATTRIBUTES*)' is only supported on: 'windows' 5.1.2600 and later.
```

### Context

- CsWin32 version: [e.g. `0.4.422-beta`]: 0.3.49-beta
- Win32Metadata version (if explicitly set by project):
- Target Framework: [e.g. `netstandard2.0`]: .NET 8
- `LangVersion` (if explicitly set by project): [e.g. `9`]: 12

**Comment by AArnott** [comment]
That looks like it's working as expected. Have you looked at the docs for CA1416 to see how to resolve it?

**Comment by myokeeh** [comment]
@AArnott, thanks. It seems I left out the part about what I've tried. I've added this to the method, but it doesn't have the desired effect of "muting" that warning.

```
[System.Runtime.Versioning.SupportedOSPlatform("windows")]
```

I had other CA1416 warnings and the above worked on those except for ones for `PInvoke.CreateHardLink()`

**Comment by AArnott** [comment]
Your attribute doesn't mention what _version_ of Windows the method requires. The warning you're seeing mentions specifically that the API is only available on 5.1.2600 and later, so your attribute needs to carry the same version requirement in order to suppress that warning.

Of course, that may have the effect of _moving_ the warning from the CsWin32 call to the call of your own method. That might be good, if your callers will include runtime version checks (which should suppress the warning automatically) before calling your method. Or I believe if you just change your TargetFramework property from `windows` to `windows5.1.2600` (or whatever the syntax is, then that effectively raises the water-level of your whole project to assume that your minimum version requirement has been met, thereby removing all warnings.
